app/controllers/users_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      sign_in @user
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  def show
+    if signed_in?
+      @user = User.find_by(id: session[:user_id])
+      @user_events = @user.events
+      @upcoming_events = @user.attended_events.upcoming
+      @past_events = @user.attended_events.past
+    else
```

Comment:
You already have a `current_user`, please use it instead of find the user with session information. In that way, you won't need the `if signed_in?` validation.